### PR TITLE
Implementing footstep sound for player

### PR DESCRIPTION
**Pull request by fm19121**
3D footstep sound for player

### CODE REVIEW

File: CSC8503/LevelManager.cpp [hunk 1]
```diff
@@ -18,10 +18,10 @@
 #include "InventoryBuffSystem/InventoryBuffSystem.h"
 #include "InventoryBuffSystem/SoundEmitter.h"
 #include "UISystem.h"
-#include "SoundManager.h"
+
 #include <filesystem>
 
-#include <fmod.hpp>
+//#include <fmod.hpp>
```
Comment (Nijelous) [inline CSC8503/LevelManager.cpp:24]: Why do we have this commented out but left in? Remove it if it's not needed.

File: CSC8503/SoundManager.cpp [hunk 2]
```diff
@@ -1,52 +1,117 @@
...
+#include "Vector3.h"
+
+using namespace NCL::CSC8503;
+using namespace NCL::Maths;
+
+SoundManager::SoundManager(GameWorld* GameWorld) {
+	mGameWorld = GameWorld;
+	//FMOD_RESULT result;
+	//FMOD::Sound* sound;
+	//FMOD::Channel* channel;
```
Comment (Nijelous) [inline CSC8503/SoundManager.cpp:11]: Again, remove the comments if they're not needed

File: CSC8503/SoundManager.cpp [hunk 3]
```diff
@@ -1,52 +1,117 @@
...
+	if (result != FMOD_OK) {
+		return;
+	}
+
+
+	//footStepSound->setMode(FMOD_3D);
+
+	//system->playSound(sound, 0, false, &channel);
+
+	//channel->setPaused(true);
```
Comment (Nijelous) [inline CSC8503/SoundManager.cpp:35]: And here

File: CSC8503/SoundManager.h [hunk 4]
```diff
@@ -1,34 +1,42 @@
...
+			//void SetSoundPosition(ISound* sound, Vector3 pos);
+
+			//void DeleteSounds();
+
+		protected:
+			GameWorld* mGameWorld;
+			FMOD::System* system = NULL;
+			FMOD::Sound* footStepSound;
+			FMOD_RESULT result;
+			FMOD::Channel* channel = nullptr;
```
Comment (Nijelous) [inline CSC8503/SoundManager.h:39]: mSystem, mFoodStepSound, mResult, mChannel

**Review by Nijelous — CHANGES_REQUESTED** [review]
The code is all sound (ha) but needs cleaning up to fit with coding standards.

With the commented code, either remove it OR add a To-Do if you plan on implementing it in the future.

**Comment by fm19121** [comment]
delete comments and fix naming

**Review by Nijelous — APPROVED** [review]

**Review by benblack90 — APPROVED** [review]

**Pull request merged**